polyfills/PageVisibility/polyfill.js

```diff
@@ -0,0 +1,17 @@
+// Page Visibility API
+(function () {
+	var prefix = document.webkitVisibilityState ? 'webkit' : document.mozVisibilityState ? 'moz' : null;
+
+	if (!prefix) {
+		return;
+	}
+
+	document.visibilityState = document[prefix + 'VisibilityState'];
+	document.hidden = document[prefix + 'Hidden'];
```

Comment:
I'd put this in a function, and call it from the event handler to avoid repeating the assignment code.